### ISSUE TITLE
[Swipe Gesture] problem onComplete works but cards do not move

### ISSUE THREAD

**Issue by Rishabh-Negi** [HeavenOSK/flutter_swipable_stack#24]
My layout is something like this

selector
_stack
__swipable stack
  
selector as data load after an api call and is passed to item count parameter

Problem: 
sometime swipe has not effect on ui cards do not move
but my onComplete function prints and shows the sipedirection.

and if i hot reload it starts working again

**Comment by HeavenOSK** [comment]
@Rishabh-Negi 
Thanks for opening this issue! I’ll investigate about the problem.
 
P.S.
Could you add a minimum reproducible code please if you can?

**Comment by Rishabh-Negi** [comment]
```
Selector<Pages, int?>(
      selector: (context, page) => page.count,
      builder: (context, itemCount, child) {
        return Scaffold(
          body: Stack(
            clipBehavior: Clip.none,
            children: [
              Column(
                children: [
                  _appBar(context),
                  Expanded(child: SizedBox()),
                  _bottomBar(),
                ],
              ),
              SwipableStack(
                itemCount: itemCount,
                onSwipeCompleted: (index, direction) {
                  print('[$index]: $direction');
                },
                onWillMoveNext: (index, direction) {
                  return direction == SwipeDirection.left ||
                      direction == SwipeDirection.right;
                },
                viewFraction: 0.98,
                builder: (context, index, constraints) {
                  var page = Provider.of<Pages>(context);
                  var doc = page.getByIndex(index);
                  if (doc.isLoading) {
                    // TODO: Check if data exists in cache or send loading data
                    return MyCard(data: doc);
                  }
                  return MyCard(data: doc);
                },
              ),
            ],
          ),
        );
      },
    )
```

for card
```
return Container(
      padding: EdgeInsets.symmetric(horizontal: 18, vertical: 10),
      margin: EdgeInsets.only(left: 30, right: 30, top: 100, bottom: 130),
      decoration: BoxDecoration(
        gradient: RadialGradient(
          colors: per > 0 ? greenCardGradient : redCardGradient,
          stops: [0.8, 1],
          radius: 1.3,
          center: Alignment.topLeft,
        ),
        boxShadow: [
          BoxShadow(offset: Offset(1, 1), blurRadius: 5, color: Colors.black26),
        ],
        borderRadius: BorderRadius.circular(12),
      ),
      child: LayoutBuilder(
        builder: (context, size) {
          return Stack(
            children: [
              Positioned(
                bottom: size.maxHeight * 0.05,
                left: 0,
                right: 5,
                top: 160,
                child: Graph(
                  points: widget.suggestions!.prices,  // FL_Chart package
                  gradientColors: color,
                ),
              ),
              Column(
                mainAxisAlignment: MainAxisAlignment.start,
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                       // list of childrens
                   ],
                 ),
            ],
          );
        },
      ),
    )
````

Problem:
> As the application starts and fetches data , I try to swipe the cards but they are freezed, if i reach max swipe threshold 
onComplete function works and show that card was swiped at that index, i we try again index is is also +1 of prev, so basically ui is not changing

Now if i hot reload it starts working sometimes

How i solved it:
> added an controller to SwipableStack, 

is it because i am not adding a controller?

**Comment by HeavenOSK** [comment]
@Rishabh-Negi 
I think it should be supported by `swipable_stack`. So I'll fix this problem in the new version 👍 
 
Maybe you can use the following workaround. You prevent showing SwipableStack before loading is completed.
 
Thanks for opening this issue.

```dart
  Selector<Pages, int?>(
    selector: (context, page) => page.count,
    builder: (context, itemCount, child) {
      return Scaffold(
        body: Stack(
          clipBehavior: Clip.none,
          children: [
            Column(
              children: [
                _appBar(context),
                Expanded(child: SizedBox()),
                _bottomBar(),
              ],
            ),
            if (itemCount != null && itemCount > 0)   // <=
              SwipableStack(
                itemCount: itemCount,
                onSwipeCompleted: (index, direction) {
                  print('[$index]: $direction');
                },
                onWillMoveNext: (index, direction) {
                  return direction == SwipeDirection.left ||
                      direction == SwipeDirection.right;
                },
                viewFraction: 0.98,
                builder: (context, index, constraints) {
                  var page = Provider.of<Pages>(context);
                  var doc = page.getByIndex(index);
                  if (doc.isLoading) {
                    // TODO: Check if data exists in cache or send loading data
                    return MyCard(data: doc);
                  }
                  return MyCard(data: doc);
                },
              ),
          ],
        ),
      );
    },
  )
```

**Comment by HeavenOSK** [comment]
@Rishabh-Negi 
Fixed this issue on latest version `0.6.2`.
 
https://pub.dev/packages/swipable_stack
 
Please try it out 🙏  
Thanks for contributing.
